workflow/validate/validate.go

```diff
@@ -149,9 +154,11 @@ func ValidateWorkflow(wftmplGetter templateresolution.WorkflowTemplateNamespaced
 	if err != nil {
 		return nil, errors.Errorf(errors.CodeBadRequest, "spec.templates%s", err.Error())
 	}
-	if ctx.Lint {
-		// if we are just linting we don't care if spec.arguments.parameters.XXX doesn't have an
-		// explicit value. workflows without a default value is a desired use case
+
+	if ctx.Lint || (ctx.WorkflowTemplateValidation && !ctx.Submit) {
```

Comment:
@tczhao are you expecting `ctx.WorkflowTemplateValidation` options in the workflow submission scenario?

Reply:
I do not believe `ctx.WorkflowTemplateValidation` will ever be `true` in a workflow submission scenario because only a `Workflow` can be submitted, not a `WorkflowTemplate`. The purpose of `(ctx.WorkflowTemplateValidation && !ctx.Submit)` is to use more permissive validation for the `Create` workflow template action and the more restrictive validation for workflow submission (if we are not linting).

Reply:
This line was refactored in https://github.com/argoproj/argo-workflows/commit/81612aec3af70c87661e33832ecd30d3084d405c.